File: explorer/API/islands/postIsland.py
import requests

def postIsland(island_json):
    url = "http://localhost:9000/islands"       
    res = requests.post(url, json= island_json)  
    return res   

#leave for testing
#postIsland({"islandId": "1", "type": 4, "name": "Mirruos", "x": 50, "y": 50,"tradegood": "2", "tradegoodTarget": "noluxury", "resourceLevel": "15", "tradegoodLevel": "11", "wonder": "1", "wonderLevel": "14", "wonderName": "הנפחייה של הפסטוס", "cityId_0": 30, "cityId_1": 6015, "cityId_2": 1967, "cityId_3": 1963, "cityId_4": 38, "cityId_5": 34, "cityId_6": 2648, "cityId_7": 936, "cityId_8": 5847, "cityId_9": 407, "cityId_10": 39, "cityId_11": 90, "cityId_12": 1290, "cityId_13": 5057, "cityId_14": 646, "cityId_15": 326, "cityId_16": 513})

File: Explorer_server/API/islands/postIsland.go
package islands

import (
	"context"
	"net/http"
	"time"

	islandsScripts "Explorer_server/DB/Scripts/islands"
	entities "Explorer_server/Entities"

	"github.com/labstack/echo/v4"
)

func postIsland(c echo.Context) (err error) {
	i := new(entities.Island)
	if err = c.Bind(i); err != nil {
		return err
	}

	island := entities.Island{
		IslandId:        i.IslandId,
		Type:            i.Type,
		Name:            i.Name,
		X:               i.X,
		Y:               i.Y,
		Tradegood:       i.Tradegood,
		TradegoodTarget: i.TradegoodTarget,
		TesourceLevel:   i.TesourceLevel,
		TradegoodLevel:  i.TradegoodLevel,
		Wonder:          i.Wonder,
		WonderLevel:     i.WonderLevel,
		WonderName:      i.WonderName,
		CityId_0:        i.CityId_0,
		CityId_1:        i.CityId_1,
		CityId_2:        i.CityId_2,
		CityId_3:        i.CityId_3,
		CityId_4:        i.CityId_4,
		CityId_5:        i.CityId_5,
		CityId_6:        i.CityId_6,
		CityId_7:        i.CityId_7,
		CityId_8:        i.CityId_8,
		CityId_9:        i.CityId_9,
		CityId_10:       i.CityId_10,
		CityId_11:       i.CityId_11,
		CityId_12:       i.CityId_12,
		CityId_13:       i.CityId_13,
		CityId_14:       i.CityId_14,
		CityId_15:       i.CityId_15,
		CityId_16:       i.CityId_16,
	}

	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()

	if err = islandsScripts.AddIsland(ctx, island); err != nil {
		return c.String(http.StatusInternalServerError, err.Error())
	}

	return c.NoContent(http.StatusCreated)
}
